src/App.jsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect, createContext } from "react";
+import "./App.css";
+import { Route, Routes, useLocation } from "react-router-dom";
+import Dashboard from "./components/Dashboard";
+
+const API_URL = "https://boolean-api-server.fly.dev/ssuihko/";
+const AppContext = createContext();
+
+function App() {
+  const [contacts, setContacts] = useState([]);
+  const [user, SetUser] = useState(null);
+  const [posts, setPosts] = useState([]);
+  const [viewPost, setViewPost] = useState([]);
+  const [viewFlag, setViewFlag] = useState(false);
+  const [viewProfile, setViewProfile] = useState(false);
+
+  const location = useLocation();
+
+  useEffect(() => {
+    if (location.pathname.includes("/profile")) {
+      setViewProfile(true);
+    }
+    if (location.pathname === "/") {
+      setViewFlag(false);
+      setViewProfile(false);
+    } else {
+      setViewFlag(true);
+    }
+  }, [location.pathname, viewProfile]);
+
+  // call data
+  useState(() => {
+    fetch(API_URL + "contact")
+      .then((response) => response.json())
+      .then((data) => {
+        setContacts(data);
+        console.log(data);
+        SetUser(data[0]);
+      });
+  }, [user]);
```

Comment:
The dependency array should be empty to only fetch on component mount (also line 46)